ratis-docs/src/site/markdown/configuraions.md

```diff
@@ -0,0 +1,563 @@
+
+| **Property**    | `raft.server.snapshot.auto.trigger.threshold`                        |
+|-----------------|----------------------------------------------------------------------|
+| **Description** | log size limit (in number of log entries) that triggers the snapshot |
+| **Type**        | long                                                                 |
+| **Default**     | 400000                                                               |
+
+| **Property**    | `raft.server.snapshot.retention.file.num` |
+|-----------------|-------------------------------------------|
+| **Description** | retentive how many old snapshot versions  |
+| **Type**        | int                                       |
+| **Default**     | -1, means only keep latest snapshot       |
+
+### DataStream
+
+ThreadPool configurations related to DataStream Api.
+
+| **Property**    | `raft.server.data-stream.async.request.thread.pool.cached` |
+|-----------------|------------------------------------------------------------|
+| **Description** | use CachedThreadPool for async request pool                |
+| **Type**        | boolean                                                    |
+| **Default**     | false                                                      |
+
+| **Property**    | `raft.server.data-stream.async.request.thread.pool.size` |
+|-----------------|----------------------------------------------------------|
+| **Description** | maximumPoolSize for async request pool                   |
+| **Type**        | int                                                      |
+| **Default**     | 32                                                       |
+
+| **Property**    | `raft.server.data-stream.async.write.thread.pool.cached` |
+|-----------------|----------------------------------------------------------|
+| **Description** | use CachedThreadPool for async write pool                |
+| **Type**        | boolean                                                  |
+| **Default**     | false                                                    |
+
+| **Property**    | `raft.server.data-stream.async.write.thread.pool.size` |
+|-----------------|--------------------------------------------------------|
+| **Description** | maximumPoolSize for async write pool                   |
+| **Type**        | int                                                    |
+| **Default**     | 16                                                     |
+
+| **Property**    | `raft.server.data-stream.client.pool.size`  |
+|-----------------|---------------------------------------------|
+| **Description** | maximumPoolSize for data stream client pool |
+| **Type**        | int                                         |
+| **Default**     | 10                                          |
+
+
+### RPC
+
+Configurations related to Server RPC timeout.
+
+| **Property**    | `raft.server.rpc.timeout.min`      |
```

Comment:
How about move these propeties related to election timeout to the `LeaderElection` chapter

Reply:
I follow the structure of `RaftServerConfigKeys`, shall we also move these configs from `Rpc` to `LeaderElection`?

Reply:
I think from the user's point of view, this configuration property is easier to find in the `LeaderElection` chapter in the documentation, and it can be placed anywhere in the code.

Reply:
Sure, I'll place these in LeaderElection only in reference then. Thanks!

Reply:
Done